enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,275 @@
+---
+title: ephemeral-agent-installer-e2e-tests
+authors:
+  - "@lranjbar"
+reviewers:
+  - "@rwsu"
+  - "@zaneb"
+  - "@pawanpinjarkar"
+approvers:
+  - "@dhellmann"
+  - "@celebdor"
+api-approvers: # in case of new or modified APIs or API extensions (CRD, aggregated api servers, webhooks, finalizers)
+  - N/A
+creation-date: 2022-02-10
+last-updated: 2022-02-15
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AGENT-20
+---
+
+# Ephemeral Agent Installer e2e Tests
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a new e2e testing framework for the new agent based installer project. 
+Currently, for this project there would be potentially two existing e2e frameworks 
+that could be used, dev-scripts and assisted-test-infra. However, both of these have 
+significant challenges in re-purposing them for this new project. 
+
+The main challenge with either of these existing frameworks is that the environment
+setup, installation and subsequent tests they run are all highly coupled together.
```

Comment:
I'm not sure this is accurate - in dev-scripts these steps are all in different scripts, the ansible roles are decoupled and used from upstream, etc - what kind of coupling did you have in mind here?

Reply:
The thing being proposed will automate the setup of infrastructure resources (libvirt, mirror registry, etc.), right? And then deploying the cluster and running any tests would be 2 steps that are independent of that setup? Maybe we need a different name than "e2e testing framework."

Reply:
I think based on the in-person conversations we've had, this paragraph can be dropped.